### PR TITLE
Branch lights nep8

### PR DESCRIPTION
**Pull request by lightszero**
实现neon nep8 功能，可以通过开关控制

### CODE REVIEW

**Comment by erikzhang** [comment]
Waiting for neo-project/neo-vm#39

File: neoj/JVM/Converter.cs [hunk 1]
```diff
@@ -44,11 +44,13 @@ public ModuleConverter(ILogger logger)
         JavaModule srcModule;
         public NeoModule outModule;
         public Dictionary<JavaMethod, NeoMethod> methodLink = new Dictionary<JavaMethod, NeoMethod>();
-        public NeoModule Convert(JavaModule _in)
+        public NeoModule Convert(JavaModule _in, ConvOption option = null)
```
Comment (shargon) [inline neoj/JVM/Converter.cs]: Is not defined on neoj, won't compile

image

File: neon/MSIL/Conv_Multi.cs [hunk 2]
```diff
@@ -724,9 +724,26 @@ private int _ConvertCall(OpCode src, NeoMethod to)
...
+                if (this.outModule.option.useNep8)
+                {
+                    int retcount = defs.ReturnType.FullName  == "System.Void" ? 0 : 1;
+
+                    byte signature = (byte)(
+                        (retcount << 7)
+                        |
+                        defs.Parameters.Count
+                        );
+                    var c = _Convert1by1(VM.OpCode.CALL_I, null, to, new byte[] { signature, 0, 0 });
```
Comment (shargon) [inline neon/MSIL/Conv_Multi.cs]: https://github.com/neo-project/neo-vm/blob/5d7985df3deaaabca8c72746144b029c7956f8d5/src/neo-vm/ExecutionEngine.cs#L942-L943

signature shouldn't need to be two bytes?

File: neoj/JVM/Converter.cs (hunk 1 above)
Comment (lightszero) [reply neoj/JVM/Converter.cs]: i think you should update neoj.csproj

File: neon/MSIL/Conv_Multi.cs (hunk 2 above)
Comment (lightszero) [reply neon/MSIL/Conv_Multi.cs]: no signatue is 1 byte,i had do some test for that,it is working fine.
it is not working for current neovm.
please try on newvm with nep8.
and if still have problem, let me known.

Comment (lightszero) [reply neon/MSIL/Conv_Multi.cs]: sorry,my bad, i missed a version of neovm.

**Comment by shargon** [comment]
It works :)

File: neoj/JVM/Converter.cs (hunk 1 above)
Comment (shargon) [reply neoj/JVM/Converter.cs]: I have the last update of this PR

**Pull request merged**